src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -460,4 +464,62 @@ default void setNotifier(Vector3i pos, @Nullable UUID uuid) {
      * @param uuid The {@link UUID} to set as notifier
      */
     void setNotifier(int x, int y, int z, @Nullable UUID uuid);
+
+    /**
+     * Gets the bounding box used to select blocks, which appears
+     * as a black outline on a vanilla client.
+     *
+     * @param pos The position of the block from which to get the selection box
+     * @return The selection box
+     */
+    default Optional<AABB> getBlockSelectionBox(Vector3i pos) {
+        return getBlockSelectionBox(pos.getX(), pos.getY(), pos.getZ());
+    }
+
+    /**
+     * Gets the bounding box used to select blocks, which appears
+     * as a black outline on a vanilla client.
+     *
+     * @param x The x coordinate of the block from which to get the selection box
+     * @param y The y coordinate of the block from which to get the selection box
+     * @param z The z coordinate of the block from which to get the selection box
+     * @return The selection box
+     */
+    Optional<AABB> getBlockSelectionBox(int x, int y, int z);
+
+    /**
+     * Gets all the block collision boxes that intersect the bounding box, in
+     * no particular order. There may be more than one box per block.
+     *
+     * @param box The intersection box
+     * @return All the intersecting block collision boxes
+     */
+    Set<AABB> getIntersectingBlockCollisionBoxes(AABB box);
+
+    /**
+     * Gets all the collision boxes that intersect the bounding box owned by
+     * the entity, in no particular order. There may be more than one box per
+     * block. This also includes entities. Will return an empty set if the
+     * owner does not have a bounding box.
+     *
+     * @param owner The entity that owns the bounding box
+     * @return All the intersecting collision boxes
+     */
+    default Set<AABB> getIntersectingCollisionBoxes(Entity owner) {
+        return owner.getBoundingBox().
```

Comment:
`checkNotNull(owner, "owner")`

dots go on beginning of next line, not end of previous line